net/mwan3/files/lib/mwan3/mwan3.sh

```diff
@@ -273,48 +276,49 @@ mwan3_set_connected_ipv4()
 
 mwan3_set_connected_iptables()
 {
-	local connected_network_v6 source_network_v6
+	local connected_network_v6
 
 	$IPS -! create mwan3_connected_v4 hash:net
 	$IPS create mwan3_connected_v4_temp hash:net
 
 	mwan3_set_connected_ipv4
 
-	$IPS -! create mwan3_connected_v6 hash:net family inet6
-	$IPS create mwan3_connected_v6_temp hash:net family inet6
+	[ $NO_IPV6 -eq 0 ] && {
+		$IPS -! create mwan3_connected_v6 hash:net family inet6
+		$IPS create mwan3_connected_v6_temp hash:net family inet6
 
-	for connected_network_v6 in $($IP6 route | awk '{print $1}' | egrep "$IPv6_REGEX"); do
-		$IPS -! add mwan3_connected_v6_temp "$connected_network_v6"
-	done
+		for connected_network_v6 in $($IP6 route | awk '{print $1}' | egrep "$IPv6_REGEX"); do
+			$IPS -! add mwan3_connected_v6_temp "$connected_network_v6"
+		done
+		$IPS swap mwan3_connected_v6_temp mwan3_connected_v6
+		$IPS destroy mwan3_connected_v6_temp
 
-	$IPS swap mwan3_connected_v6_temp mwan3_connected_v6
-	$IPS destroy mwan3_connected_v6_temp
+		$IPS -! create mwan3_source_v6 hash:net family inet6
+		$IPS create mwan3_source_v6_temp hash:net family inet6
+		for source_network_v6 in $($IP6 addr ls  | sed -ne 's/ *inet6 \([^ \/]*\).* scope global.*/\1/p'); do
+			$IPS -! add mwan3_source_v6_temp "$source_network_v6"
+		done
+		$IPS swap mwan3_source_v6_temp mwan3_source_v6
+		$IPS destroy mwan3_source_v6_temp
+	}
 
 	$IPS -! create mwan3_connected list:set
 	$IPS -! add mwan3_connected mwan3_connected_v4
-	$IPS -! add mwan3_connected mwan3_connected_v6
-
-	$IPS -! create mwan3_source_v6 hash:net family inet6
-	$IPS create mwan3_source_v6_temp hash:net family inet6
-	for source_network_v6 in $($IP6 addr ls  | sed -ne 's/ *inet6 \([^ \/]*\).* scope global.*/\1/p'); do
-		$IPS -! add mwan3_source_v6_temp "$source_network_v6"
-	done
-	$IPS swap mwan3_source_v6_temp mwan3_source_v6
-	$IPS destroy mwan3_source_v6_temp
+	[ $NO_IPV6 -eq 0 ] && $IPS -! add mwan3_connected mwan3_connected_v6
 
 	$IPS -! create mwan3_dynamic_v4 hash:net
 	$IPS -! add mwan3_connected mwan3_dynamic_v4
 
-	$IPS -! create mwan3_dynamic_v6 hash:net family inet6
-	$IPS -! add mwan3_connected mwan3_dynamic_v6
+	[ $NO_IPV6 -eq 0 ] && $IPS -! create mwan3_dynamic_v6 hash:net family inet6
+	[ $NO_IPV6 -eq 0 ] && $IPS -! add mwan3_connected mwan3_dynamic_v6
 }
 
 mwan3_set_general_rules()
 {
 	local IP
 
 	for IP in "$IP4" "$IP6"; do
-
+		[ "$IP" = "$IP6" ] && [ $NO_IPV6 -eq 1 ] && continue
```

Comment:
Some times I see `[ $NO_IPV6 -eq 1 ]` or `[ $NO_IPV6 = 0 ]` we should unify the check or is there any reason for this?

Reply:
I did it for code concision. Sometimes it is easier to only run code if ipv6 is turned on, sometimes it is more concise to do an early bailout if ipv6 is not enabled.

For example, here we could change it to:
if [ "$IP" = "$IP6" ] && [ $NO_IPV6 -eq 0 ]; then
...
fi

However what I wrote is a bit cleaner. Generally if it is in a for loop, I check if we are in the IPV6 branch, and bail. If it is more linear code, I check if IPV6 is on, and then run the IPV6 code if IPV6 is on.

Reply:
That's not what I meant at all. I wanted to know why `-eq and sometimes `=`

Reply:
Yes, should be "-eq" everywhere. Can you point me to where it is '=' and I will fix?

I'm not seeing it at all 
```
$ grep NO_IPV6 -R
files/lib/mwan3/mwan3.sh:NO_IPV6=$?
files/lib/mwan3/mwan3.sh:	[ $NO_IPV6 -eq 0 ] && {
files/lib/mwan3/mwan3.sh:	[ $NO_IPV6 -eq 0 ] && $IPS -! add mwan3_connected mwan3_connected_v6
files/lib/mwan3/mwan3.sh:	[ $NO_IPV6 -eq 0 ] && $IPS -! create mwan3_dynamic_v6 hash:net family inet6
files/lib/mwan3/mwan3.sh:	[ $NO_IPV6 -eq 0 ] && $IPS -! add mwan3_connected mwan3_dynamic_v6
files/lib/mwan3/mwan3.sh:		[ "$IP" = "$IP6" ] && [ $NO_IPV6 -eq 1 ] && continue
files/lib/mwan3/mwan3.sh:		[ "$IPT" = "$IPT6" ] && [ $NO_IPV6 -eq 1 ] && continue
files/lib/mwan3/mwan3.sh:	elif [ "$family" = "ipv6" ] && [ $NO_IPV6 -eq 0 ]; then
files/lib/mwan3/mwan3.sh:		[ $NO_IPV6 -eq 1 ] && return
files/lib/mwan3/mwan3.sh:	if [ "$family" = "ipv6" ] && [ $NO_IPV6 -eq 0 ]; then
files/lib/mwan3/mwan3.sh:	if [ "$family" = "ipv6" ] && [ $NO_IPV6 -eq 0 ]; then
files/lib/mwan3/mwan3.sh:	if [ "$family" = "ipv6" ] && [ $NO_IPV6 -eq 0 ]; then
files/lib/mwan3/mwan3.sh:	elif [ "$family" = "ipv6" ] && [ $NO_IPV6 -eq 0 ] && [ $is_offline -eq 0 ]; then
files/lib/mwan3/mwan3.sh:		[ "$IPT" = "$IPT6" ] && [ $NO_IPV6 -eq 1 ] && continue
files/lib/mwan3/mwan3.sh:				[ "$IPT" = "$IPT6" ] && [ $NO_IPV6 -eq 1 ] && continuea
files/lib/mwan3/mwan3.sh:					[ "$IPT" = "$IPT6" ] && [ $NO_IPV6 -eq 1 ] && continue
files/lib/mwan3/mwan3.sh:					[ "$IPT" = "$IPT6" ] && [ $NO_IPV6 -eq 1 ] && continue
files/lib/mwan3/mwan3.sh:					[ "$IPT" = "$IPT6" ] && [ $NO_IPV6 -eq 1 ] && continue
files/lib/mwan3/mwan3.sh:			[ "$IPT" = "$IPT6" ] && [ $NO_IPV6 -eq 1 ] && continue
files/lib/mwan3/mwan3.sh:		[ "$IPT" = "$IPT6" ] && [ $NO_IPV6 -eq 1 ] && continue
files/usr/sbin/mwan3:	[ $NO_IPV6 -eq 1 ] && return
files/usr/sbin/mwan3:	[ $NO_IPV6 -eq 1 ] && return
files/usr/sbin/mwan3:	[ $NO_IPV6 -eq 1 ] && return
files/usr/sbin/mwan3:		[ "$IP" = "$IP6" -a $NO_IPV6 -eq 1 ] && continue
files/usr/sbin/mwan3:		[ "$IPT" = "$IPT6" -a $NO_IPV6 -eq 1 ] && continue
```

Reply:
LGTM

Reply:
Ah - actually I know it's not what you meant with testing 0 vs 1, but you were actually correct. The test for ip6tables won't necessarily return '1' whet it fails, so we should test against 0 in all cases. I will update this commit with the change.

Reply:
okay - fixed that in:
https://github.com/openwrt/packages/pull/12814/commits/30a46bdc9e409134c95c7e7760b438ef6f1dbdd0 and 
https://github.com/openwrt/packages/pull/12814/commits/a0d66d4eebefe6e89c582df2b4dc972ea7c5c7f5